python-package/lightgbm/sklearn.py

```diff
@@ -532,8 +532,9 @@ def set_params(self, **params):
         """
         for key, value in params.items():
             setattr(self, key, value)
-            if hasattr(self, '_' + key):
-                setattr(self, '_' + key, value)
+            under_score = "_"
+            if hasattr(self, f"{under_score} {key}"):
+                setattr(self, f"{under_score} {key}", value)
```

Comment:
```suggestion
            if hasattr(self, f"_{key}"):
                setattr(self, f"_{key}", value)
```

I don't think it's necessary to pull underscore out into its own variable. f-strings can be a mix of literal strings and Python expressions.

Also note that the `+` operator for strings does not add a space. So the equivalent of `'_' + key` is `f"_{key}"`, not `f"_ {key}"`